src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/RoleAssignmentFilter.java

```diff
@@ -90,7 +90,12 @@ private RoleAssignmentFilter() {
             for (PermissionTypes type : permissionsRequired) {
                 permissionPredicates.add(builder.isTrue(taskRoleResources.get(type.value().toLowerCase(Locale.ROOT))));
             }
-            final Predicate permissionPredicate = builder.or(permissionPredicates.toArray(new Predicate[0]));
+            Predicate permissionPredicate;
+            if (andPermissions) {
+                permissionPredicate = builder.and(permissionPredicates.toArray(new Predicate[0]));
+            } else {
+                permissionPredicate = builder.or(permissionPredicates.toArray(new Predicate[0]));
+            }
```

Comment:
As discussed, we can probably improve this in another ticket with some refactoring.  Maybe think of a way of expressing the relationship between permissions.  We could provide an operand or provide a conjuntion/disjunctive specific objects.  One to think about.

Reply:
Fixed.